addons/gamification/models/gamification_karma_tracking.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from odoo import api, fields, models
+
+
+class KarmaTracking(models.Model):
+    _name = 'gamification.karma.tracking'
+    _description = 'Track Karma Changes'
+
+    user_id = fields.Many2one('res.users', 'User', required=True)
+    old_value = fields.Integer('Old Karma Value', required=True, readonly=1)
+    new_value = fields.Integer('New Karma Value', required=True, readonly=1)
+    consolidation_type = fields.Selection(selection=[('none', 'Not Consolidated'),
+                                                     ('daily', 'Daily Consolidated'),
+                                                     ('monthly', 'Monthly Consolidated'),
+                                                     ('yearly', 'Yearly Consolidated')],
+                                          default='none')
+    tracking_date = fields.Datetime(default=fields.Datetime.now)
+
+    @api.model
+    def _consolidate_last_day(self):
+        to_date = datetime.now()
+        from_date = to_date - relativedelta(days=1)
+        self._process_consolidate(from_date, to_date, 'none', 'daily', '%Y-%m-%d')
+
+    @api.model
+    def _consolidate_last_month(self):
```

Comment:
Actually rethinking a bit the consolidation, I wonder if we should not only keep a monthly consolidation. Indeed a month -> year consolidation means cutting by 12 which is not really important (tables should handle even thousands of students with 12 entries / year).

I think we should handle things like one week after beginning of month: consolidate previous month. That way we have a single cron and consolidation is done with one method.

It seems it will simplify this diff (less crons, less methods). What do you think ?